app/controllers/hub/state_file/automated_messages_controller.rb

```diff
@@ -1,26 +1,71 @@
 module Hub::StateFile
   class AutomatedMessagesController < Hub::StateFile::BaseController
+    include StateFile::SurveyLinksConcern
+    helper_method :email_message
+    helper_method :sms_body
 
     def index
-      @messages = messages_preview
+      Rails.application.eager_load!
+      @messages = StateFile::AutomatedMessage::BaseAutomatedMessage.descendants
+      @locales = [:en, :es]
+      @us_state = StateFile::StateInformationService.active_state_codes.include?(params[:us_state]) ? params[:us_state] : "az"
+      get_intake
     end
 
     private
 
-    def messages_preview
-      Rails.application.eager_load!
-      message_classes = StateFile::AutomatedMessage::BaseAutomatedMessage.descendants
-      message_classes.to_h do |klass|
-        replaced_body = klass.new.email_body.gsub('<<', '&lt;&lt;').gsub('>>', '&gt;&gt;')
-        email = StateFileNotificationEmail.new(to: "example@example.com",
-                                               body: replaced_body,
-                                               subject: klass.new.email_subject)
-        message = StateFile::NotificationMailer.user_message(notification_email: email)
-        ActionMailer::Base.preview_interceptors.each do |interceptor|
-          interceptor.previewing_email(message)
-        end
-        [klass, message]
+    def get_intake
+      return @intake if @intake.present?
+      state_class = "StateFile#{@us_state.titleize}Intake".constantize
+      if params[:intake_id].present?
+        @intake = state_class.find_by_id(params[:intake_id])
+        return @intake if @intake.present?
+        flash[:alert] = "Unknown Intake"
       end
+      @intake = state_class.new(
+        locale: "en",
+        primary_first_name: "Cornelius"
+      )
+    end
+
+    def message_params
+      intake = get_intake
+      state_code = intake.state_code
+      locale = intake.locale || "en"
+      submitted_key = intake.efile_submissions.count > 1 ? "resubmitted" : "submitted"
+      {
+        primary_first_name: intake.primary_first_name,
```

Comment:
so every time there is a new message we'll have to add message params here? We have a method called `base_args` in the `app/services/state_file/messaging_service.rb` and as you used here the survey links are in `StateFile::SurveyLinksConcern`. I wonder if we could everything in one place so we won't have to remember to put it in different places every time

Reply:
Good call - I'm gonna do this and we can re-review

Reply:
So I added additional specs here to test the automated-messages endpoint. It loads all descendants of the `BaseAutomatedMessage`, so if you add one that has additional parameters it should cause the spec to fail until they are added to this hash.